src/main/kotlin/com/demonwav/mcdev/buildsystem/BuildSystem.kt

```diff
@@ -32,27 +24,24 @@ import java.util.ArrayList
  */
 abstract class BuildSystem {
 
-    private val lock = Any()
+    val dependencies = mutableSetOf<BuildDependency>()
+    val repositories = mutableSetOf<BuildRepository>()
```

Comment:
Is this really fine as a `Set`? This order is kind of relevant for Maven repositories because it determines which repositories are checked first.